### PR TITLE
5.0.2_medex

### PR DESCRIPTION
**Pull request by ophthal**

### CODE REVIEW

File: library/MedEx/API.php [hunk 1]
```diff
@@ -950,13 +1024,31 @@ public function generate($token, $events)
...
                                     ORDER BY cal.pc_eventDate,cal.pc_startTime";
-                $result = sqlStatement($sql_GOGREEN, $escapedArr);
-
+              //  $this->MedEx->logging->log_this($sql_GOGREEN);
+              //  $this->MedEx->logging->log_this($escapedArr);
+                try {
+                    $result = sqlStatement($sql_GOGREEN, $escapedArr);
+                } catch (\Exception $e) {
+                     $dd = print_r($escapedArr, true);
+                    echo "hello<pre>$sql_GOGREEN\n".$dd."\n".$e."\n";
```
Comment (bradymiller) [inline library/MedEx/API.php]: I'm assuming above is a debug thing and should be commented out

File: library/MedEx/API.php [hunk 2]
```diff
@@ -1054,27 +1137,21 @@ private function addRecurrent($appt, $interval, $timing, $timing2, $M_group = "R
...
-        $this->MedEx->logging->log_this($hits);
-        $this->MedEx->logging->log_this("above are the hits for pc_eid: " . $appt['pc_eid']."\n\n
-        &&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&\n
-        Now Convert them to real appt slots for MedEx\n");
-
+        
         //any dates that match need to be spawned from recurrent and made to live on their own.
-        $oldRecurrspec = unserialize($appt['pc_recurrspec'], ['allowed_classes' => false]);
-
+        $oldRecurrspec = unserialize($appt['pc_recurrspec']);
```
Comment (bradymiller) [inline library/MedEx/API.php]: revert above line change or is a big security vulnerability (let me know if not working with the change)

File: library/MedEx/API.php [hunk 3]
```diff
@@ -1218,92 +1289,31 @@ public function calculateEvents($event, $start_date, $stop_date)
...
-                    if ($excluded == false) {
-                        $data[] = $occurence;
+            switch ($event['pc_recurrtype']) {
+                //not recurrent
+                case '0':
+                    $data[] = $event;
+                    break;
+                case '1':
+                case '3':
+                    $event_recurrspec = @unserialize($event['pc_recurrspec']);
```
Comment (bradymiller) [inline library/MedEx/API.php]: revert above line change or is a big security vulnerability (let me know if not working with the change)

File: library/MedEx/API.php [hunk 4]
```diff
@@ -1314,24 +1324,82 @@ public function calculateEvents($event, $start_date, $stop_date)
...
-                    $occurenceYmX = date('Y-m-d', mktime(0, 0, 0, $nm+$rfreq, $nd, $ny));
-                    list($ny,$nm,$nd) = explode('-', $occurenceYmX);
-                    $occurenceYm = "$ny-$nm";
-                }
-                break;
-        }
+                    break;
+    
+                    case '2':
+                    $event_recurrspec = @unserialize($event['pc_recurrspec']);
```
Comment (bradymiller) [inline library/MedEx/API.php]: revert above line change or is a big security vulnerability (let me know if not working with the change)

File: library/MedEx/API.php [hunk 5]
```diff
@@ -2222,7 +2287,7 @@ function SMS_bot(pid) {
                 window.open('<?php echo $GLOBALS['webroot']; ?>/interface/main/messages/messages.php?nomenu=1&go=SMS_bot&pid=' + pid,'SMS_bot', 'width=370,height=600,resizable=0');
                 return false;
             }
-            $(function() {
+            $(document).ready(function() {
```
Comment (bradymiller) [inline library/MedEx/API.php]: revert above line change (this was changed throughout entire codebase)

File: library/MedEx/API.php [hunk 6]
```diff
@@ -2896,15 +2961,15 @@ public function display_add_recall($pid = 'new')
 
             </div>
             <script>
-                $(function () {
+                $(document).ready(function () {
```
Comment (bradymiller) [inline library/MedEx/API.php]: revert above line change (this was changed throughout entire codebase)

File: library/MedEx/API.php [hunk 7]
```diff
@@ -3333,7 +3413,7 @@ function validatePassword (pw, options) {
                 // great success!
                 return true;
             }
-            $(function() {
+            $(document).ready(function() {
```
Comment (bradymiller) [inline library/MedEx/API.php]: revert above line change (this was changed throughout entire codebase)

**Comment by bradymiller** [comment]
hi @ophthal , I just reviewed this code.

File: library/MedEx/API.php (hunk 1 above)
Comment (ophthal) [reply library/MedEx/API.php]: Yup.
Fix coming...

File: library/MedEx/API.php (hunk 2 above)
Comment (ophthal) [reply library/MedEx/API.php]: No problemo.

**Comment by bradymiller** [comment]
hi @ophthal , Code looks good and bringing this in. Thanks for the fixes! Planning any more code changes prior to the 5.0.2 release?(I am about to branch the code, but still will take in code for another 1-2 weeks)
-brady

**Comment by ophthal** [comment]
Yes.  I have Eye Form Updates...

**Comment by bradymiller** [comment]
hi @opthal ,
Is there a PR with the eye form updates, or will that be coming soon?
thanks, -brady

**Comment by ophthal** [comment]
Very soon for the Eye Form - by Sunday.

I get the impression that the mobile stuff is not worth submitting.

**Comment by sjpadgett** [comment]
Hi @ophthal , If you can make eye module work on small devices by all means. My issues come from when we start interfering with desktop to make mobile work.
Encounter form is probably the most useful place to use small devices depending how it is done ie if scaling a normally desktop to small device or acting as a client to forms backend.
I'd be interested in how your doing it.

**Comment by ophthal** [comment]
I'll put up the branch for a lark.  It is not Eye Form specific.  It covers the document interface (upload/download/view), the calendar, the flow board and SMS interface.   The idea is to inspire a real programmer to consider adding this functionality to the codebase, using a framework.

**Comment by ophthal** [comment]
The [two demo](https://two.openemr.io/ is currently running off my MOE branch. It has a simple bootstrap based wrapper for mobile device detection to add limited functionality.  Currently I use this for HIPAA compliant image/video uploads and to check next day’s schedule.  

Please try this from different devices.

**Pull request merged**